packages/bot-web-ui/src/components/contract-result-overlay/contract-result-overlay.spec.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { render, screen } from '@testing-library/react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import RootStore from 'Stores/index';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import ContractResultOverlay from './contract-result-overlay';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({
+    saveRecentWorkspace: jest.fn(),
+    unHighlightAllBlocks: jest.fn(),
+}));
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+
+const mock_ws = {
+    authorized: {
+        subscribeProposalOpenContract: jest.fn(),
+        send: jest.fn(),
+    },
+    storage: {
+        send: jest.fn(),
+    },
+    contractUpdate: jest.fn(),
+    subscribeTicksHistory: jest.fn(),
+    forgetStream: jest.fn(),
+    activeSymbols: jest.fn(),
+    send: jest.fn(),
+};
+describe('ContractResultOverlay', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+
+    beforeEach(() => {
+        jest.resetModules();
+        const mock_store = mockStore({});
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('should render the ContractResultOverlay component', () => {
+        const { container } = render(<ContractResultOverlay profit={0} className={''} />, {
+            wrapper,
+        });
+        expect(container).toBeInTheDocument();
+    });
+
+    it('should show contract won', () => {
+        render(<ContractResultOverlay profit={0} className={''} />, {
+            wrapper,
+        });
+        expect(screen.getByText('Won')).toBeInTheDocument();
```

Comment:
just a question, this looks fine to me, but should we also check if the necessary classname is present 
db-contract-card__result--won

Reply:
@vinu-deriv i think checking for `won` or `lost` should be enough in this case as it will render the unique text based on the profit value.

Reply:
yeah should be fine